tests/foreman/cli/test_discoveryrule.py

```diff
@@ -136,51 +178,50 @@ def test_positive_create_with_priority(self):
 
         @Assert: Rule should be successfully created and has expected priority
         value
-
-        @caseautomation: notautomated
         """
+        rule_priority = '100'
+        rule = self._make_discoveryrule({u'priority': rule_priority})
+        self.assertEqual(rule['priority'], rule_priority)
 
     @run_only_on('sat')
-    @stubbed
     @tier1
     def test_positive_create_disabled_rule(self):
         """Create Discovery Rule in disabled state
 
         @id: 8837a0c6-e19a-4c33-8b87-07b6f69dbb0f
 
         @Assert: Disabled rule should be successfully created
-
-        @caseautomation: notautomated
         """
+        rule = self._make_discoveryrule({u'enabled': 'false'})
+        self.assertEqual(rule['enabled'], 'false')
 
     @run_only_on('sat')
-    @stubbed
     @tier1
     def test_negative_create_with_invalid_name(self):
         """Create Discovery Rule with invalid names
 
         @id: a0350dc9-8f5b-4673-be88-a5e35d1f8ca7
 
         @Assert: Error should be raised and rule should not be created
-
-        @caseautomation: notautomated
         """
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(CLIFactoryError):
+                    self._make_discoveryrule({'name': name})
 
     @run_only_on('sat')
-    @stubbed
     @tier1
     def test_negative_create_with_invalid_hostname(self):
         """Create Discovery Rule with invalid hostname
 
         @id: 0ae51085-30d0-44f9-9e49-abe928a8a4b7
 
         @Assert: Error should be raised and rule should not be created
-
-        @caseautomation: notautomated
         """
+        with self.assertRaises(CLIFactoryError):
+            self._make_discoveryrule({u'hostname': '$#@!*'})
```

Comment:
What kind of validation is present for hostname? Is hammer checking connection to that host or it just validates the name by some rules? In first case your test is ok, in the second case - i'd suggest using some custom datafactory helper which will cover cjk, utf8, html, boundaries etc.